File: 1721-swapping-nodes-in-a-linked-list/1721. Swapping Nodes in a Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getLength(const ListNode* root) {
        int size = 0;;
        while (root) {
            root = root->next;
            size++;
        }
        return size;
    }
    ListNode* swapNodes(ListNode* head, int k) {
        int n = getLength(head);
        if (n == 1) return head;
        ListNode* a = nullptr;
        ListNode* b = nullptr;
        ListNode* curr = head;
        cout << n-k << endl;
        for (int i = 0; i < n; i++) {
            if (i == k-1) {
                a = curr;
            }  
            if (i == n-k) {
                cout << "reached" << endl;
                b = curr;
            }
            curr = curr->next;
        }
        int temp = a->val;
        cout << b->val << "bitch" << endl;
        a->val = b->val;
        b->val = temp;
        return head;
    }
};

File: 1721-swapping-nodes-in-a-linked-list/1721. Swapping Nodes in a Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        def length(curr):
            size = 0
            while curr:
                size += 1
                curr = curr.next
            return size
        n = length(head)
        if n == 1: return head
        count = 0
        temp = None
        temp2 = None
        curr = head
        while curr:
            if count == k-1:
                temp = curr
            if count == n-k:
                temp2 = curr    
            curr = curr.next
            count += 1
        temp.val, temp2.val = temp2.val, temp.val
        return head
            
        